### ISSUE TITLE
Multiscreen support?

### ISSUE THREAD

**Issue by copyme** [faho/kwin-tiling#88]
Hey! Does the script supports multiscreen configurations? I see that there is a repo called mutliscreen but the last commit is kinda old. Well, I have noticed some problems with two screen and I wonder if I should install the script from that repo or submit a bug report.

**Comment by faho** [comment]
The "multimonitor" branch (which I think you mean) was merged. 

My personal usage of multiple monitors (occasionally connecting to a TV) works. Anything in particular that doesn't for you?

**Comment by copyme** [comment]
@faho Yes you are right. I meant the "multimonitor" branch. Well, sometimes on the second screen I have an empty space (right tiling column is empty) even though there is only one window on the screen at all. If I try to re-scale such a window by hand then it re-scales back -- so I cannot use this empty space for anything. So far, I found that re-logging to a session solves the problem but it is not a nice solution. If it will happen once again I will try to make a video demonstrating the issue. 

BTW doesn't know if this matters but it only has happened on the second virtual desktop, i.e. I use 2 virtual desktops and two screens.

**Comment by faho** [comment]
Which version of the script are you using?

Do you use any other kwin scripts?

**Comment by copyme** [comment]
@faho I use the current git master version. The other scripts which I use are: "Enforces Window Decorations on GTK+ window" and "Minimize Others".

**Comment by faho** [comment]
Okay, so we'd need to narrow down when this happens.

Does this happen with specific programs (e.g. only with chrome or dolphin or gtk-things?)?

Does this happen after you do something specific (e.g. use "Minimize Others"? Quit a program? Open a new window?)?

**Comment by copyme** [comment]
It just happened again. 
![crash_tiles](https://user-images.githubusercontent.com/456607/30102953-74ed5418-92f1-11e7-886e-345fc27a3910.png) (see the hole between the windows)

I have tried to run programs and see when this happens but so far I had no lack. I will keep you updated.

**Comment by copyme** [comment]
OK I got it. 

1. Open some windows on the virtual desktop 1
2. Open some windows on the virtual desktop 2
3. Go to the virtual desktop 1 and switch a window to "Move to All Desktops"
4. Go to the virtual desktop 2 and place the widow with "Move to All Desktops" on to the screen 2 together with another window

Right now, the tiling layout on the virtual desktop 1 (screen 1) is broken, i.e. it contains "holes".

The steps probably, could be simplified. I will keep you updated.

**Comment by copyme** [comment]
Today, I got a similar problem while working only with one virtual desktop and two screens. I have two machines and the problem happens only while working on the one with two screens. So I guess we have a bug in the support. I will try to find the steps to reproduce the issue.

**Comment by copyme** [comment]
Today, I got a similar problem while working only with one virtual desktop and two screens. I have two machines and the problem happens only while working on the one with two screens. So I guess we have a bug in the support. I will try to find the steps to reproduce the issue.

**Comment by copyme** [comment]
OK. The steps are:

1. Place some windows on the screen 2.
2. Place some windows on the screen 1.
3. Active a windows which is on the screen 2
4. Use Meta+Shift + L (or H) to move it to another column.

**Comment by copyme** [comment]
I found another but related issue. When an application on start draws its window on the second screen then sometimes a place for it is "allocated" on the first screen. 

On the screen shot we have that Mathematica appeared on the second screen but a place for it was assigned on the first screen. This led to a "hole" in the right tiling column of the first screen. 
![tiles_problem](https://user-images.githubusercontent.com/456607/30166103-993d6190-93e2-11e7-99ca-61a5da12e22c.png)

**Comment by ilkerhk** [comment]
This happens to me also. However, I think Mathematica windows have something different, especially the palette. Sometimes after monitor connect disconnect holes appear. Not sure if it will help in your case but  I bind a keyboard shortcut to the following script which fixes my issues (it simply restarts kwin and traverses all desktops so kwin-tiling rearranges the windows).  
best, 

$cat restartKwin.sh 

(kwin --replace &) 
sleep 3
currDesk=$(wmctrl -d | grep "*" | awk '{print $1}')
for desk in $(wmctrl -d | grep -v "*" |  awk '{print $1}'); do  
         wmctrl -s "$desk"
done
wmctrl -s "$currDesk"

**Comment by copyme** [comment]
@ilkerhk Thanks! Unfortunately, when I ran your script twice the second time kwin ran without all my scripts, and the only way to fix this was to re-login. Any idea what is wrong?

**Comment by ilkerhk** [comment]
For some reason the the asteriks sign doesnt show up in the code above. That might be the reason. There is asteriks in between double quotes, I can see when I edit in github but when I submit comment it dissapears. 
I tried to put it in code tag but now newlines doesn't show. Can you try with asterisk sign between double quotes in two grep calls?

**Comment by copyme** [comment]
@ilkerhk still running kwin_x11 --replace multiple times gets rid of scripts until next login. Not even restarting plasmashell helps. I guess there is a bug in Plasma 5.8.

**Comment by ilkerhk** [comment]
Do you mean when you run "kwin --replace"  the kwin-tiling script stops working?  

I would check if tiling works on new windows after restart.   First run "kwin --replace" 
then check if tiling works on newly opened windows (for example start new instances of konsole or dolphin etc)

if that is the case than Faho fixed it see :   https://github.com/faho/kwin-tiling/issues/90
(you need to install the latest code.)

**Comment by copyme** [comment]
@ilkerhk Indeed, it works after pulling up the changes from the repo. Thank you! The other problem is that the other scripts that I use are not adapted and they do not work after kwin_x11 --replace. I will contact their authors.
